### PR TITLE
chore: Add test readme[WIP]

### PR DESCRIPTION
**Pull request by kevjumba**
Signed-off-by: Kevin Zhang <kzhang@tecton.ai>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+### To include a new offline / online store in the main Feast repo
+
+* Extend `data_source_creator.py` for your offline store.
+* In `repo_configuration.py` add a new`IntegrationTestRepoConfig` or two (depending on how many online stores you want to test).
+* Run the full test suite with `make test-python-integration.`
+
+### Including a new offline / online store in the main Feast repo from external plugins with community maintainers.
+
+* This folder is for plugins that are officially maintained with community owners. Place the APIs in feast/infra/offline_stores/contrib/.
```
Comment (achals) [inline sdk/python/tests/README.md:166]:
```suggestion
* This folder is for plugins that are officially maintained with community owners. Place the APIs in `feast/infra/offline_stores/contrib/`.
```

File: sdk/python/tests/README.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+* Extend `data_source_creator.py` for your offline store and implement the required APIs.
+* In `contrib_repo_configuration.py` add a new `IntegrationTestRepoConfig` (depending on how many online stores you want to test).
+* Run the test suite on the contrib test suite with `make test-python-contrib-universal`.
+
+### To include a new online store
+
+* In `repo_configuration.py` add a new config that maps to a serialized version of configuration you need in `feature_store.yaml` to setup the online store.
+* In `repo_configuration.py`, add new`IntegrationTestRepoConfig` for offline stores you want to test.
```
Comment (achals) [inline sdk/python/tests/README.md:174]:
```suggestion
* In `repo_configuration.py`, add new`IntegrationTestRepoConfig` for online stores you want to test.
```

File: sdk/python/tests/README.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+Starting 6001
+Starting 6002
+Starting 6003
+Starting 6004
+Starting 6005
+Starting 6006
+~~~~
+* You should be able to run the integration tests and have the redis cluster tests pass.
+* If you would like to run your own redis cluster, you can run the above commands with your own specified ports and connect to the newly configured cluster.
+* To stop the cluster, run `./create-cluster stop` and then `./create-cluster clean`.
```
Comment (achals) [inline sdk/python/tests/README.md:211]: nit but this is a strange command (`stop`) to pass into a script called `create-cluster`. Maybe we should call it `redis-cluster` or something?

File: sdk/python/tests/README.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    fs.apply([fv, entity])
+
+    # ... run test
+```
+
+### Running your own redis cluster for testing
+
+* Install redis on your computer. If you are a mac user, you should be able to `brew install redis`.
+    * Running `redis-server --help` and `redis-cli --help` should show corresponding help menus.
+* Run `cd scripts/create-cluster` and run `./create-cluster start` then `./create-cluster create` to start the server. You should see output that looks like this:
```
Comment (achals) [inline sdk/python/tests/README.md:200]:
```suggestion
* Run `./infra/scripts/create-cluster.sh start` then `./infra/scripts/create-cluster.sh create` to start the redis cluster locally. You should see output that looks like this:
```

File: sdk/python/tests/README.md (hunk 3 above)
Comment (achals) [inline sdk/python/tests/README.md:211]:
```suggestion
* To stop the cluster, run `./infra/scripts/create-cluster.sh stop` and then `./infra/scripts/create-cluster.sh clean`.
```

File: sdk/python/tests/README.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+### To test a new offline / online store from a plugin repo
+
+* Install Feast in editable mode with `pip install -e`.
+* The core tests for offline / online store behavior are parametrized by the `FULL_REPO_CONFIGS` variable defined in `feature_repos/repo_configuration.py`. To overwrite this variable without modifying the Feast repo, create your own file that contains a `FULL_REPO_CONFIGS` (which will require adding a new `IntegrationTestRepoConfig` or two) and set the environment variable `FULL_REPO_CONFIGS_MODULE` to point to that file. Then the core offline / online store tests can be run with `make test-python-universal`.
+* See the [custom offline store demo](https://github.com/feast-dev/feast-custom-offline-store-demo) and the [custom online store demo](https://github.com/feast-dev/feast-custom-online-store-demo) for examples.
+
+### To include a new offline / online store in the main Feast repo
+
+* Extend `data_source_creator.py` for your offline store.
+* In `repo_configuration.py` add a new`IntegrationTestRepoConfig` or two (depending on how many online stores you want to test).
```
[the remark this answers is not in the document]
Comment (kevjumba) [reply sdk/python/tests/README.md:161]: fixed.

File: sdk/python/tests/README.md (hunk 1 above)
Comment (kevjumba) [reply sdk/python/tests/README.md]: fixed

File: sdk/python/tests/README.md (hunk 2 above)
Comment (kevjumba) [reply sdk/python/tests/README.md]: fixed

File: sdk/python/tests/README.md (hunk 4 above)
Comment (kevjumba) [reply sdk/python/tests/README.md]: fixed

File: sdk/python/tests/README.md (hunk 3 above)
Comment (kevjumba) [reply sdk/python/tests/README.md]: Sure let me change

Comment (kevjumba) [reply sdk/python/tests/README.md]: fixed

**Review by kevjumba — COMMENTED** [review]
I just realized the other integration test readme pr hasn't merged yet, I am going to close this and add all of these changes to the other pr.

**Pull request closed without merging**